Mathlib/Order/SuccPred/Basic.lean

```diff
@@ -979,13 +979,13 @@ namespace WithTop
 
 section Succ
 
-variable [DecidableEq α] [PartialOrder α] [OrderTop α] [SuccOrder α]
+variable [Preorder α] [@DecidablePred α IsMax] [SuccOrder α]
```

Comment:
Please just remove `[@DecidablePred α IsMax]` and use `classical` or `open scoped Classical in`. 
This predicate will only be rarely provable (e.g. only on finite types with decidable le)

Reply:
This is actually provable in a partial order with decidable equality, thanks to `succ_eq_iff_isMax`. Should I just use that?

Reply:
Ah, good point. I'm fine with either option.